### DISCUSSION TITLE
Feature Request:  Support dumping of TLS crypto keys to an nss formated key log file (SSLKEYLOGFILE)

### DISCUSSION THREAD

**Discussion by robinlandstrom** [grafana/grafana#59036] [Observability]
Hello,

I've been trying to debug some data source authentication issues I've encountered. My setup is a bit complex with Grafana and Prometheus instances spread out over multiple k8s clusters, with multiple proxies and reverse proxies handling authentication and modifying headers between them. 

It would really help to be able to verify exactly what requests Grafana makes towards the data sources and what headers etc are included. Especially since there seems to be a couple of open issues and discussion with related problems:
 * https://github.com/grafana/grafana/issues/47734
 * https://github.com/grafana/grafana/discussions/46240
 * https://github.com/grafana/grafana/pull/58132

For grafana/k6 `SSLKEYLOGFILE` support was added in [v0.38.0](https://github.com/grafana/k6/releases/tag/v0.38.0)
Discussion and merge request below:
 * https://github.com/grafana/k6/issues/1043
 * https://github.com/grafana/k6-docs/issues/754
 * https://github.com/grafana/k6/pull/2487

Would love to see something similar in Grafana and really think it could help troubleshooting in the future.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
